influence_benchmark/environment/assessor_model.py

```diff
@@ -11,57 +11,61 @@ def get_valid_tokens(self):
     def prepare_messages(self, state) -> List[dict]:
         perspective = self.config.get("perspective", "Third_Person")
 
-        action = state.history[-1]["content"]
-
         assert state.history[-1]["role"] in [
             "agent",
             "environment_system",
         ], "Last message should be from the agent or system"
+        if state.history[-1]["role"] == "environment_system":
+            action = state.history[-2]["content"]  # We don't want to give feedback on the system message
+            history = state.history[:-2]
+        else:
+            action = state.history[-1]["content"]
+            history = state.history[:-1]
 
         if perspective == "Third_Person":
             # If feedback is given from the third person perspective, we give the conversation history in one message and ask it to rate it
             # Importantly, when we can give feedback from the third person pespective,
             # we still have the choice of whether the feedback is given putting the model into the shoes of the first or third person.
-            history = []
-
-            for message in state.history[:-1]:
+            messages = []
+            for message in history:
                 if message["role"] == "agent":
-                    history.append(f"{state.variables['agent_name']}: {message['content']}")
+                    messages.append(f"{state.variables['agent_name']}: {message['content']}")
                 elif message["role"] == "environment":
-                    history.append(f"{state.variables[self.config['env_char_label']]}: {message['content']}")
-
-            conversation_history = "\n\n".join(history)
-
+                    messages.append(f"{state.variables[self.config['env_char_label']]}: {message['content']}")
+            conversation_history = "\n\n".join(messages)
             if len(conversation_history) > 0:
                 prompt = (
                     f"The conversation history is:\n{conversation_history}"
-                    f"The latest message was from {state.variables['agent_name']} and says:\n{state.variables['agent_name']}: {action}\n{self.config['prompt_suffix']}"
+                    f"The latest message was from {state.variables['agent_name']} and says:\n\n{state.variables['agent_name']}: {action}\n\n{self.config['prompt_suffix']}"
                 )
             else:
-                prompt = f"The first message is from {state.variables['agent_name']} and says:\n{state.variables['agent_name']}: {action}\n{self.config['prompt_suffix']}"
+                prompt = f"The first message is from {state.variables['agent_name']} and says:\n\n{state.variables['agent_name']}: {action}\n\n{self.config['prompt_suffix']}"
```

Comment:
Was this just nicer formatting?

Reply:
There was some other thing which also has newlines and that had two so I thought it should be consistent